doc/sphinx/architecture.md

```diff
@@ -0,0 +1,252 @@
+- If one is only interested in implementing their own *Mir*-based compositor, they need not look any further than this directory.
+
+## `/src`
+- The `src` folder at the root of the project comprises the core implementation of *Mir*.
+
+## `/examples`
+- A collection of demo *Mir* compositors that are used as examples of how one might build a true compositor. Examples include:
+  - `miral-app`
+  - `miral-shel`
+  - `mir_demo_server`
+
+## `/tests`
+- Unit tests for the entire project.
+
+# Libraries
+The following is the library tree:
+```
+├── miral
+├── miroil
+├── mircore
+├── mircommon
+├── mircookie
+└── mirserver
+    ├── mircompositor
+    ├── mirinput
+    ├── mirfrontend
+    ├── mirfrontend-wayland
+    ├── mirfrontend-xwayland
+    ├── mirshell
+    ├── mirshelldecoration
+    ├── mirscene
+    ├── mirgraphics
+    ├── mirreport
+    ├── mirlttng
+    ├── mirlogging
+    ├── mirnullreport
+    ├── mirconsole
+    ├── mirgl
+    └── mirrenderergl
+├── mirplatform
+├── mirwayland
+├── graphics-gbm-kms
+├── graphics-eglstreamkms
+├── graphics-x11
+├── graphics-wayland
+├── input-evdev
+└── renderer-egl-generic
+```
+
+## miral
+- The "Mir Abstraction Layer" (aka `MirAL` or `miral`) is an API that makes Mir easy for compositor-implementers to work with. It provides the core window management concepts and an interface that is more ABI stable than Mir's internal mirserver API.
+- Public-facing
+- Directories:
+  - `include/miral`
+  - `src/miral`
+  
+## miroil
+- A custom *Mir* library for the [Lomiri](https://lomiri.com/) folks
+- This is unimportant for 99% of people
+- Directories:
+  - `include/miroil`
+  - `src/miroil`
+
+## mircore
+- Fundamental data structures like 2D points, Rectangles, File Descriptor, etc.
+- Public-facing
+- Directories:
+  - `include/core`
+  - `src/core`
+  -
+## mircommon
+- Data structures and utility functions like Mir-event building, logging, keymaps, etc.
+- Differs from `mircore` in that the concepts map strictly onto *Mir*'s concepts instead of being general concepts like a Rectangle
+- Public-facing
+- Directories:
+  - `include/common`
+  - `src/common`
+  - `src/include/common`
+
+## mircookie
+- Provides event timestamps for inputs events that are difficult to spoof.
+- Public-facing
+- Directories:
+  - `include/cookie`
+  - `src/cookie`
+  - `src/include/cookie`
+
+## mirserver
+- Provides the core *Mir* runtime, with the entry-point being the `mir::Server` class.
+- The `Server` class has a `DisplayServer` which provides access to the interfaces that that support the server's runtime.
+  - The `ServerConfiguration` provides a bunch of methods called `the_XYZ()` to access these various instances, where "XYZ" can be replaced with the interface of interest.
+- The classes that the `Server` relies on for its runtime can be found organized in separate folders in this directory.
+- Directories:
+  - `include/server`
+  - `src/server`
+  - `src/include/server`
+  
+## mircompositor
```

Comment:
I don't like this naming (probably comes from trying to treat it at the same level of abstraction as libraries).

It is "just" a namespace within mirserver.

Reply:
It could be `mirserver::mircompositor`?

Reply:
That's worse! How is a reader to associate something in the code (`mir::compositor::Foo`, say) with `mirserver::mircompositor`?